libs/base/System/Clock.idr

```diff
@@ -77,7 +77,7 @@ showTime s ns (MkClock seconds nanoseconds) =
       nanoseconds = show (cast nanoseconds `div` quotient)
   in concat [ padLeft s '0' seconds
             , "."
-            , padRight ns '0' nanoseconds
+            , padRight ns '0' (padLeft (min ns 9) '0' nanoseconds)
```

Comment:
Isn't the right approach something like
```
padRight ns '0' $ take ns $ padLeft 9 '0' nanoseconds
```
?

Reply:
I see, I wasn't thorough enough in my tests, drat. Certainly your change is the correct one given more test cases. Will update with further corrections.